include/irr/asset/filters/dithering/CPrecomputedDither.h

```diff
@@ -0,0 +1,89 @@
+// This file is part of the "IrrlichtBAW" engine.
+// For conditions of distribution and use, see copyright notice in irrlicht.h
+
+#ifndef __IRR_C_PRECOMPUTED_DITHER_H_INCLUDED__
+#define __IRR_C_PRECOMPUTED_DITHER_H_INCLUDED__
+
+#include "../include/irr/asset/filters/dithering/CDither.h"
+
+namespace irr
+{
+	namespace asset
+	{
+		//! A class to apply dithering to an image using precomputed dithering image
+		/*
+			
+		*/
+
+		class CPrecomputedDither : public CDither<CPrecomputedDither>
+		{
+			public:
+				CPrecomputedDither() {}
+				virtual ~CPrecomputedDither() {}
+
+				//! State of precomputed dithering class
+				/*
+					The state requires only input dithering image
+					which buffer will be used in dithering process
+					in full extent of image.
+				*/
+				
+				class CState : public CDither::CState
+				{
+					public:
+						CState(const asset::ICPUImage* const ditheringImage) 
+						{
+							ditherImageData.buffer = ditheringImage->getBuffer();
```

Comment:
you can't just take over a buffer of an image (even if you could you should have constructed a smartrefctd ptr)

because the buffer might not be there or cover too small of an area.

You need the CopyImageToBuffer filter IIRC.

Reply:
or create a proxy image with its own buffer and flatten the input image into that.

Reply:
what?

Reply:
do we have such?

Reply:
https://github.com/buildaworldnet/IrrlichtBAW/blob/shader_pipeline/include/irr/asset/filters/CFlattenRegionsImageFilter.h

Should be a flatten & convert or you should just only accept images in the RGBA UNORM16 or UNORM32 formats

Reply:
done, but not sure if I should still force RGBA UNORM16 or UNORM32 formats